packages/workspace/src/browser/workspace-frontend-contribution.ts

```diff
@@ -42,7 +42,13 @@ export class WorkspaceFrontendContribution implements CommandContribution, Keybi
     registerCommands(commands: CommandRegistry): void {
         commands.registerCommand(WorkspaceCommands.OPEN, {
             isEnabled: () => true,
-            execute: () => this.doOpen()
+            execute: (args: any[]) => {
+                if (args) {
+                    const [fileURI] = args;
+                    return this.workspaceService.open(fileURI);
```

Comment:
I had a merge conflict here, so I ask: why do not we have this functionality in the `WorkspaceCommands.OPEN_WORKSPACE` command? What do you think? Also, if the URI represents a file, what should happen? Thanks!

Reply:
In general I think we should rethink the way commands are written. Quick open stuffs should only provide the UI interactions, and when writting a command we should pass the function to be executed by a quick open.
I would like that we have a proper test framework like what I suggest here: https://github.com/theia-ide/theia/issues/3160 before we start such a refactoring.

Reply:
I agree, having more tests would be great, and we can work on the commands but still, don't you think your modification should go to `OPEN_WORKSPACE` instead of `OPEN`? :)

Reply:
I can take care of it if we decide to move this from `OPEN` to `OPEN_WORKSPACE`.